### DISCUSSION TITLE
Properly Chunk Loading

### DISCUSSION THREAD

**Discussion by GoldenGamerLP** [Minestom/Minestom#675] [Help] [answered]
Is there a good way loading Chunks? Im only getting Chunk X: 0 Z: 0 to load? What should i do?

This is my current code:
![image](https://user-images.githubusercontent.com/65727384/153771679-911af0b6-4b57-4aee-a2d2-e5a45b519752.png)

This is the debug:
![image](https://user-images.githubusercontent.com/65727384/153771694-5603922b-4e9a-48b9-8179-8382bc7f3e7a.png)

Note:
- It should also load Chunks (-1, 0), (0, -1), (-1,-1)

**Comment by MrlnHi** [comment]
change `x < upper.chunkX()` to `x <= upper.chunkX()`, same for z

**Reply by GoldenGamerLP** [reply]
I did not work, it only loaded the chunks with positiv numbers:
- 
![image](https://user-images.githubusercontent.com/65727384/153773274-08c95329-4045-4665-897a-dbd1a97109eb.png)


`public CompletableFuture<Instance> loadChunks(Instance instance, Schematic schematic) {
        return CompletableFuture.supplyAsync(() -> {
            Pos lower = Pos.ZERO.add(0,32,0);
            Pos upper = Pos.ZERO.add(0, 32, 0).add(schematic.getWidth(),schematic.getHeight(),schematic.getLength());
            for(int x = lower.chunkX(); x <= upper.chunkX(); x++) {
                for(int z = lower.chunkZ(); z <= upper.chunkZ(); z++) {
                    LOGGER.info("Loading: x, z: -> " + x+" "+z);
                    instance.loadChunk(x,z).join();
                }
            }
            return instance;
        });
    }`

**Reply by MrlnHi** [reply]
well that is because you add the schematic dimensions onto Pos.ZERO

**Reply by GoldenGamerLP** [reply]
Yes, i oriented myself on the code which the schematic libary uses.
- 
![image](https://user-images.githubusercontent.com/65727384/153773557-2491c65f-e87e-4a51-9bc6-7f6b8c6c0e80.png)

Also, im pasting the schematic at Pos.Zero.add(0,32,0);

**Reply by MrlnHi** [reply]
Why would your schematic go into the negatives?

**Reply by GoldenGamerLP** [reply]
Chunks can be negativ?

**Reply by MrlnHi** [reply]
I know that, I just don't see why a schematic's origin wouldn't be at its min x y z (but that's besides the point, idk how to help you anymore)

**Comment by GoldenGamerLP** [comment]
I dont know, i saw it with the f3 chunk debugging, that the schematic did go into negatives

![image](https://user-images.githubusercontent.com/65727384/153773714-96c191de-b18b-4494-adf7-c3fad4cb1bfe.png)

Edit: What should i do then? The Errors that the Chunks arent loaded come anyways

**Comment by GoldenGamerLP** [comment, marked as answer]
I figured out how to do it:
- Without an Java Doc its quit difficult but here is my code which works. Advice: You should place a worldborder to prevent players into going to unloaded chunks.

![image](https://user-images.githubusercontent.com/65727384/153918341-49367e35-33c7-46f1-890d-4e5e346a8af3.png)
